lms/static/js/learner_dashboard/views/course_entitlement_view.js

```diff
@@ -0,0 +1,398 @@
+                         return;
+                     }
+                     enrollBtnInitial.removeClass('disabled');
+
+                     // Update button text specifying if the user is initially enrolling, changing or leaving a session.
+                     if (newSessionId) {
+                         enrollText = currentSessionId ? gettext('Change Session') : gettext('Select Session');
+                     } else {
+                         enrollText = gettext('Leave Current Session');
+                     }
+                     enrollBtnInitial.text(enrollText);
+                     this.removeDialog(enrollBtnInitial);
+                     this.initializeVerificationDialog(enrollBtnInitial);
+                 },
+
+                 toggleSessionSelectionPanel: function() {
+                     /*
+                     Opens and closes the session selection panel.
+                     */
+                     this.$el.toggleClass('hidden');
+                     if (!this.$el.hasClass('hidden')) {
+                         // Set focus to the session selection for a11y purposes
+                         this.$('.session-select').focus();
+                         this.hideDialog(this.$('.enroll-btn-initial'));
+                     }
+                     this.updateEnrollBtn();
+                 },
+
+                 initializeVerificationDialog: function(invokingElement) {
+                     /*
+                     Instantiates an instance of the Bootstrap v4 dialog modal and attaches it to the passed in element.
+
+                     This dialog acts as the second step in verifying the user's action to select, change or leave an
+                     available course session.
+                      */
+                     var confirmationMsgTitle,
+                         confirmationMsgBody,
+                         popoverDialogHtml,
+                         currentSessionId = this.entitlementModel.get('currentSessionId'),
+                         newSessionId = this.$('.session-select').find('option:selected').data('session_id');
+
+                     // Update the button popover text to enable two step authentication.
+                     if (newSessionId) {
+                         confirmationMsgTitle = !currentSessionId ?
+                             gettext('Are you sure you want to select this session?') :
+                             gettext('Are you sure you want to change to a different session?');
+                         confirmationMsgBody = !currentSessionId ? '' :
+                             gettext('Any course progress or grades from your current session will be lost.');
+                     } else {
+                         confirmationMsgTitle = gettext('Are you sure that you would like to leave this session?');
+                         confirmationMsgBody = gettext('Any course progress or grades from your current session will ' +
+                                'be lost.');
```

Comment:
Can we move this onto one line on line 287? I'm not sure how the extraction scripts will handle it as is.

Reply:
Yeah sure